app/Filament/Resources/UserResource/Pages/ListUsers.php

```diff
@@ -154,6 +157,14 @@ public function table(Table $table): Table
                     ->multiple()
                     ->searchable()
                     ->preload(),
+                SelectFilter::make('roles')
+                    ->label('Roles')
+                    ->options($this->getRolesOption())
+                    ->getSearchResultsUsing(fn (string $search) => Role::query()->where(new Expression('lower(name)'), 'like', '%' . strtolower($search) . '%')->take(50)->pluck('name', 'id')->toArray())
+                    ->query(fn (Builder $query, array $data) => $this->roleFilter($query, $data))
```

Comment:
The same as `options()`, can you please remove the extra method and do the filtering in here so that the filter is more easily reusable.